File: test-local.ts
/**
 * 本地测试脚本 - 直接测试搜索功能
 * 
 * 运行: npx ts-node test-local.ts
 * 或: node dist/test-local.js (需要先编译)
 */

import { AISearcher } from "./src/searcher.js";

async function main() {
  const query = process.argv[2] || "什么是 MCP Model Context Protocol";
  const language = process.argv[3] || "zh-CN";

  console.log(`\n🔍 测试搜索: "${query}"`);
  console.log(`📍 语言: ${language}\n`);

  const searcher = new AISearcher();

  try {
    const result = await searcher.search(query, language, false);

    if (result.success) {
      console.log("✅ 搜索成功!\n");
      console.log("=".repeat(60));
      console.log("AI 回答:");
      console.log("=".repeat(60));
      console.log(result.aiAnswer.substring(0, 2000));
      if (result.aiAnswer.length > 2000) {
        console.log("\n... (内容已截断)");
      }
      console.log("\n" + "=".repeat(60));
      console.log(`来源链接 (${result.sources.length} 个):`);
      console.log("=".repeat(60));
      for (const source of result.sources) {
        console.log(`- ${source.title}`);
        console.log(`  ${source.url}\n`);
      }
    } else {
      console.log("❌ 搜索失败:", result.error);
    }
  } catch (error) {
    console.error("❌ 错误:", error);
  } finally {
    await searcher.close();
    console.log("\n🔚 测试完成");
  }
}

main();

File: test_local.py
"""本地测试脚本"""
import asyncio
import sys
sys.path.insert(0, 'src')
from google_ai_search import AsyncGoogleAISearcher

async def test():
    print("=" * 60)
    print("本地测试 - AsyncGoogleAISearcher（无头模式）")
    print("=" * 60)
    
    # 使用无头模式（headless=True）
    # 通过反检测参数（--disable-blink-features=AutomationControlled）绕过 Google 检测
    searcher = AsyncGoogleAISearcher(timeout=60, headless=True, use_user_data=True)
    result = await searcher.search('Python 是什么编程语言？', language='zh-CN')
    
    print(f"Success: {result.success}")
    print(f"AI Answer length: {len(result.ai_answer)}")
    print(f"Sources count: {len(result.sources)}")
    
    if result.ai_answer:
        preview = result.ai_answer[:300] + "..." if len(result.ai_answer) > 300 else result.ai_answer
        print(f"\nAI Answer preview:\n{preview}")
    
    if result.error:
        print(f"\nError: {result.error}")
    
    if result.sources:
        print(f"\nSources:")
        for i, src in enumerate(result.sources[:3], 1):
            print(f"  {i}. {src.title} - {src.url[:50]}...")
    
    await searcher.close_session()
    print("\n" + "=" * 60)

if __name__ == "__main__":
    asyncio.run(test())
